ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -1,11 +1,13 @@
 {-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveGeneric              #-}
```

Comment:
Are these LANGUAGE changes necessary? EG I didn't see any overloaded strings used here 🤔

Reply:
`OverloadedStrings` is used here: https://github.com/input-output-hk/ouroboros-network/blob/c9551452bffc1fdcd004ed27f0d7aa87c7ab7f0b/ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs#L702

Reply:
`DeriveGeneric` is used here: https://github.com/input-output-hk/ouroboros-network/blob/eb5e8563960437bb74ce271cd473f9b0ba39ee90/ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs#L678

Reply:
Without `DuplicateRecordFields` I get this error:

```
src/Ouroboros/Consensus/Shelley/Ledger/Query.hs:294:19: error:
    Not in scope: ‘_pstakeMark’
    |
294 |                 { _pstakeMark
    |                   ^^^^^^^^^^^

src/Ouroboros/Consensus/Shelley/Ledger/Query.hs:295:19: error:
    Not in scope: ‘_pstakeSet’
    |
295 |                 , _pstakeSet
    |                   ^^^^^^^^^^

src/Ouroboros/Consensus/Shelley/Ledger/Query.hs:296:19: error:
    Not in scope: ‘_pstakeGo’
    |
296 |                 , _pstakeGo
    |                   ^^^^^^^^^
```